x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/indicator_match/threat_mapping/create_threat_signals.ts

```diff
@@ -169,14 +171,83 @@ export const createThreatSignals = async ({
     let list = await getDocumentList({ searchAfter: undefined });
     let documentCount = totalDocumentCount;
 
+    // this is re-assigned depending on max clause count errors
+    let chunkPage = itemsPerSearch;
+
     while (list.hits.hits.length !== 0) {
       verifyExecutionCanProceed();
-      const chunks = chunk(itemsPerSearch, list.hits.hits);
+      const chunks = chunk(chunkPage, list.hits.hits);
       ruleExecutionLogger.debug(`${chunks.length} concurrent indicator searches are starting.`);
       const concurrentSearchesPerformed =
         chunks.map<Promise<SearchAfterAndBulkCreateReturnType>>(createSignal);
+
       const searchesPerformed = await Promise.all(concurrentSearchesPerformed);
-      results = combineConcurrentResults(results, searchesPerformed);
+
+      // Did our searches fail with an error containing the maxClauseCount
+      // error message?
+
+      // TODO - delete this comment
+      // FOR PR TESTING PURPOSES ONLY
+      // uncomment this line..
+      // const maxClauseCountValue = MAX_CLAUSE_DEFAULT_VALUE;
+
+      // and comment out the below reduce function to force the
+      // max clause count error to re-appear.
+      const maxClauseCountValue = searchesPerformed.reduce<number>((acc, search) => {
+        const failureMessage: string | undefined = search.errors.find((err) =>
+          err.includes('failed to create query: maxClauseCount is set to')
+        );
+
+        const regex = /[0-9]/g;
+        const foundMaxClauseCountValue = failureMessage?.match(regex);
+
+        if (!isEmpty(foundMaxClauseCountValue) && foundMaxClauseCountValue != null) {
+          const tempVal = parseInt(foundMaxClauseCountValue.join(''), 10);
+          let val = MAX_CLAUSE_DEFAULT_VALUE;
+          if (tempVal === 0) {
+            // if the resulting max clause count error is 0, don't subtract 1 from that.
+            // rare case that really should not happen unless ES claims heap size is 0.
+            // ref: https://github.com/elastic/elasticsearch/issues/86136
```

Comment:
How can we get a max clause count of 0? Even if the heap size is 0 isn't the max clause count 1024?

Reply:
yup you're correct i'm going to delete this logic too.